File: auth/src/main/java/com/harrish/auth/exception/InvalidTokenException.java
package com.harrish.auth.exception;

import com.harrish.auth.exception.error.AuthErrorCode;

public class InvalidTokenException extends BaseException {

    public InvalidTokenException() {
        super(AuthErrorCode.AUTH_INVALID_TOKEN.getMessageKey(),
                "Invalid token");
    }

    public static InvalidTokenException expired() {
        return new InvalidTokenException(AuthErrorCode.AUTH_TOKEN_EXPIRED.getMessageKey(),
                "Token has expired");
    }

    private InvalidTokenException(String errorCode, String message) {
        super(errorCode, message);
    }
}

File: auth-kotlin/src/main/kotlin/com/harrish/auth/exception/InvalidTokenException.kt
package com.harrish.auth.exception

import com.harrish.auth.exception.error.AuthErrorCode

class InvalidTokenException : BaseException(
    AuthErrorCode.AUTH_INVALID_TOKEN.messageKey,
    "Invalid token"
) {
    companion object {
        fun expired(): InvalidTokenException {
            return InvalidTokenException(AuthErrorCode.AUTH_TOKEN_EXPIRED.messageKey, "Token has expired")
        }
    }

    private constructor(errorCode: String, message: String) : super(errorCode, message)
}
